ipatests/test_integration/test_acme.py

```diff
@@ -266,3 +266,78 @@ def test_centralize_acme_disable(self):
         """Test if ACME disable on replica if disabled on master"""
         result = self.master.run_command(['ipa-acme-manage', 'status'])
         assert 'disabled' in result.stdout_text
+
+
+@pytest.mark.skipif(not cainstance.minimum_acme_support(),
+                    reason="does not provide ACME")
+class TestACMECALess(IntegrationTest):
+    """Test to check the CA less replica setup"""
+    num_replicas = 1
+    num_clients = 1
+
+    #########################################
+    #                TODO                   #
+    # refactor install after PR#5147 merged #
+    #########################################
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install packages before client install in case of IPA DNS problems
+        if not skip_certbot_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'certbot'])
+        if not skip_mod_md_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'mod_md'])
+
+        tasks.install_master(cls.master, setup_dns=True)
+
+        tasks.install_client(cls.master, cls.clients[0])
+        tasks.config_host_resolvconf_with_master_data(
+            cls.master, cls.clients[0]
+        )
+        tasks.install_replica(cls.master, cls.replicas[0], setup_ca=False)
+
+    def test_cafull_centralize_acme_enable(self):
+        """Test if ACME is enabled after CA-less to CA-full deployment
+
+        Deployment where one server is deployed as CA-less, when converted
+        to CA full, should have ACME enabled by default.
+
+        related: https://pagure.io/freeipa/issue/8524
+        """
+        tasks.kinit_admin(self.master)
+        # enable acme on master
+        self.master.run_command(['ipa-acme-manage', 'enable'])
+
+        # wait for acme to get enable
+        exc = None
+        for _i in range(5):
+            time.sleep(2)
+            try:
+                self.clients[0].run_command(
+                    ['curl', '--fail', self.acme_server])
+                break
+            except Exception as e:
+                exc = e
+        else:
+            raise exc
```

Comment:
I don't understand why this block comes before checking the status, which does the same thing.

From a client perspective you don't know which master you're getting. Since ipa-ca should be a CNAME it's likely that you're pinging the master and replica in round-robin.